### ISSUE TITLE
Generate big pdf ( 250 pages) format pdf A-2B : print to PDF: Printing failed (-32000)

### ISSUE THREAD

**Issue by QuentinChaumont** [gotenberg/gotenberg#446]
We use gotenberg to generate pdf format A-2B using chromium but failed for big files.

The process is working for small pdf (30 pages). But when we want to generate bigger one (250 pages and 1000 pages) the process failed.

Generate big files without format pdf A-2B is also working.

Request example : 
`curl \
--request POST '[http://localhost:3000/forms/chromium/convert/html'](http://localhost:3000/forms/chromium/convert/html%27) \
--form 'files=@"/path/to/index.html"' \
--form 'pdfFormat="PDF/A-2b"' \
-o my.pdf`

_Error gotenberg_ : ` "convert HTML to PDF: convert to PDF: chromium PDF: print to PDF: Printing failed (-32000)"`

The error seems to come from :
[ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory 

_Debug stack  :_
> {"level":"debug","ts":1649754891.2555325,"logger":"api.formschromiumconverthtml","msg":"print to PDF with: &{Landscape:false DisplayHeaderFooter:true PrintBackground:false Scale:1 PaperWidth:16.54166 PaperHeight:11.7 MarginTop:0 MarginBottom:0 MarginLeft:0 MarginRight:0 PageRanges: IgnoreInvalidPageRanges:false HeaderTemplate:<html><head></head><body></body></html> FooterTemplate:<html><head></head><body></body></html> PreferCSSPageSize:false TransferMode:}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754891.2555625,"logger":"api.formschromiumconverthtml.browser","msg":"-> {\"id\":22,\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\",\"method\":\"Page.printToPDF\",\"params\":{\"displayHeaderFooter\":true,\"scale\":1,\"paperWidth\":16.54166,\"paperHeight\":11.7,\"marginTop\":0,\"marginBottom\":0,\"marginLeft\":0,\"marginRight\":0,\"headerTemplate\":\"\\u003chtml\\u003e\\u003chead\\u003e\\u003c/head\\u003e\\u003cbody\\u003e\\u003c/body\\u003e\\u003c/html\\u003e\",\"footerTemplate\":\"\\u003chtml\\u003e\\u003chead\\u003e\\u003c/head\\u003e\\u003cbody\\u003e\\u003c/body\\u003e\\u003c/html\\u003e\"}}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754893.8353238,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.styleSheetAdded\",\"params\":{\"header\":{\"styleSheetId\":\"48.1\",\"frameId\":\"3BF1DD850143A1F5055149B685D19D45\",\"sourceURL\":\"\",\"origin\":\"regular\",\"title\":\"\",\"ownerNode\":10,\"disabled\":false,\"isInline\":false,\"isMutable\":false,\"isConstructed\":false,\"startLine\":0,\"startColumn\":0,\"length\":54,\"endLine\":0,\"endColumn\":54}},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7671938,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7672672,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7672853,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7673156,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7673314,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7673461,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"Page.frameResized\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7673676,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7673826,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7673972,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.7674253,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754951.767442,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.mediaQueryResultChanged\",\"params\":{},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649754953.041991,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"Page.lifecycleEvent\",\"params\":{\"frameId\":\"3BF1DD850143A1F5055149B685D19D45\",\"loaderId\":\"B6AE9B3040A45125A1EC443694E88125\",\"name\":\"InteractiveTime\",\"timestamp\":82911.294613},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649755007.3219748,"logger":"api.formschromiumconverthtml.browser","msg":"[0412/091647.321770:ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq: No such file or directory (2)\n[0412/091647.321890:ERROR:file_io_posix.cc(144)] open /sys/devices/system/cpu/cpu0/cpufreq/scaling_max_freq: No such file or directory (2)\n","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649755007.3305554,"logger":"api.formschromiumconverthtml.browser","msg":"[0412/091647.325938:ERROR:directory_reader_posix.cc(42)] opendir /tmp/Crashpad/attachments/1cab1e5d-fe5f-4dbb-9a5d-3a333432b83d: No such file or directory (2)\n","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649755008.0098526,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"id\":22,\"error\":{\"code\":-32000,\"message\":\"Printing failed\"},\"sessionId\":\"30D2DCD8648B911761EB9A11204181F4\"}","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649755008.010061,"logger":"api.formschromiumconverthtml","msg":"remove user profile directory '/tmp/71e43cb7-d7af-4d96-8835-49f2164ee763'","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> {"level":"debug","ts":1649755008.030683,"logger":"api.formschromiumconverthtml","msg":"'/tmp/08443c30-f29b-4d0e-8fc5-0a8260d72d19' removed","trace":"e0aea925-1ddd-4760-b910-7b915b8e75ac"}
> 

Maybe it's same problem than here : https://bugs.chromium.org/p/chromium/issues/detail?id=1228625

**Comment by gulien** [comment]
Hello @QuentinChaumont

Actually, Chromium do the first conversion (HTML to "standard" PDF format), then LibreOffice converts the resulting file to the `PDF/A-2B` format.

Are you sure it works if the `pdfFormat` form field is not filled? As mentioned previously, it's two separate processes 😄

**Comment by gulien** [comment]
@QuentinChaumont a workaround would be to split your conversion in chunks (can be done in parallel), merge the resulting PDFs, then convert the merged PDFs to `PDF/A-2B`.

**Comment by QuentinChaumont** [comment]
Thanks for your answers. 

We'll try your workaround.

**Comment by QuentinChaumont** [comment]
Update: We managed to generate big pdf (over 2000 pages) using Gotenberg 6. Gotenberg 7 failed for big pdf. 
No Update for big pdf conversion on format A-2B using gotenberg. 
Waiting for a fix.

**Comment by gulien** [comment]
@QuentinChaumont so Gotenberg 6´s Chrome version works for big file 🤔 please note it’s a long running process in v6; there are memory leaks.

Not sure what’s the fix here, but to wait for a new version of Chrome that fixes the issue.

**Comment by QuentinChaumont** [comment]
Yes Gotenberg 6 successfully worked to convert to pdf using **/convert/html**. (Thanks for the memory leaks details, we'll monitoring it, but no other choice now).
Same html is not converting on Gotenberg 7 using **/forms/libreoffice/convert** or **/forms/chromium/convert/html** (with or without using format A-2B).

**Comment by bambuca** [comment]
The problem is still there, please reopen.

**Comment by Dhopal** [comment]
Hello, I'm facing the same issue with gotenberg 7. 
Did we find any fix for gotenberg 7 or should I use gotenberg 6?

**Comment by gulien** [comment]
@Dhopal not yet

**Comment by noureentaj** [comment]
@Dhopal @gulien I am facing this issue in the latest version of Gotenberg: 7.8.0
And this is very urgent, I'd really appreciate any help I can get on this, thanks in advance!
I also would very much appreciate any help on what are the suggested configurations for production. I am auto-scaling my page width and since the PDF size is too huge, I get a File not found error.
 
Here are the Logs:

`{"level":"debug","ts":1675761799.2159505,"logger":"api.formschromiumconverthtml","msg":"event LoadEventFired fired","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761799.2161229,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"id\":21,\"result\":{\"root\":{\"nodeId\":6,\"backendNodeId\":2,\"nodeType\":9,\"nodeName\":\"#document\",\"localName\":\"\",\"nodeValue\":\"\",\"childNodeCount\":2,\"children\":[{\"nodeId\":7,\"parentId\":6,\"backendNodeId\":5,\"nodeType\":10,\"nodeName\":\"html\",\"localName\":\"\",\"nodeValue\":\"\",\"publicId\":\"\",\"systemId\":\"\"},{\"nodeId\":8,\"parentId\":6,\"backendNodeId\":6,\"nodeType\":1,\"nodeName\":\"HTML\",\"localName\":\"html\",\"nodeValue\":\"\",\"childNodeCount\":2,\"children\":[{\"nodeId\":9,\"parentId\":8,\"backendNodeId\":7,\"nodeType\":1,\"nodeName\":\"HEAD\",\"localName\":\"head\",\"nodeValue\":\"\",\"childNodeCount\":2,\"attributes\":[]},{\"nodeId\":10,\"parentId\":8,\"backendNodeId\":8,\"nodeType\":1,\"nodeName\":\"BODY\",\"localName\":\"body\",\"nodeValue\":\"\",\"childNodeCount\":3,\"attributes\":[]}],\"attributes\":[\"lang\",\"en\"],\"frameId\":\"4CF50E3412634D2BE175F3F92FD721CD\"}],\"documentURL\":\"file:///tmp/2c16e6f3-1167-4644-9fe0-09c3f139ff0a/index.html\",\"baseURL\":\"file:///tmp/2c16e6f3-1167-4644-9fe0-09c3f139ff0a/index.html\",\"xmlVersion\":\"\",\"compatibilityMode\":\"NoQuirksMode\"}},\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\"}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.2155993,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"Page.lifecycleEvent\",\"params\":{\"frameId\":\"4CF50E3412634D2BE175F3F92FD721CD\",\"loaderId\":\"FA03499E8F7BAEECBF23561FAC5B397A\",\"name\":\"networkAlmostIdle\",\"timestamp\":5196880.316588},\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\"}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.215654,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"Page.lifecycleEvent\",\"params\":{\"frameId\":\"4CF50E3412634D2BE175F3F92FD721CD\",\"loaderId\":\"FA03499E8F7BAEECBF23561FAC5B397A\",\"name\":\"firstMeaningfulPaint\",\"timestamp\":5196879.301439},\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\"}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.2156742,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"Page.lifecycleEvent\",\"params\":{\"frameId\":\"4CF50E3412634D2BE175F3F92FD721CD\",\"loaderId\":\"FA03499E8F7BAEECBF23561FAC5B397A\",\"name\":\"networkIdle\",\"timestamp\":5196880.316588},\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\"}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.2157836,"logger":"api.formschromiumconverthtml","msg":"event networkIdle fired","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.2158298,"logger":"api.formschromiumconverthtml","msg":"default white background not hidden","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.215888,"logger":"api.formschromiumconverthtml.browser","msg":"-> {\"id\":22,\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\",\"method\":\"Runtime.evaluate\",\"params\":{\"expression\":\"\\n(() =\\u003e {\\n\\tconst css = 'html { -webkit-print-color-adjust: exact !important; }';\\n\\n\\tconst style = document.createElement('style');\\n\\tstyle.type = 'text/css';\\n\\tstyle.appendChild(document.createTextNode(css));\\n\\tdocument.head.appendChild(style);\\n})();\\n\",\"returnByValue\":true}}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.5904374,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"DOM.childNodeCountUpdated\",\"params\":{\"nodeId\":9,\"childNodeCount\":3},\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\"}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.5905256,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"id\":22,\"result\":{\"result\":{\"type\":\"undefined\"}},\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\"}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.5905921,"logger":"api.formschromiumconverthtml","msg":"no extra link tags","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.5906014,"logger":"api.formschromiumconverthtml","msg":"no emulated media type","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.5906103,"logger":"api.formschromiumconverthtml","msg":"no extra script tags","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.5906162,"logger":"api.formschromiumconverthtml","msg":"no wait delay","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.590622,"logger":"api.formschromiumconverthtml","msg":"no wait for window status nor wait for expression","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.590629,"logger":"api.formschromiumconverthtml","msg":"no custom header nor footer","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.5906632,"logger":"api.formschromiumconverthtml","msg":"print to PDF with: &{Landscape:false DisplayHeaderFooter:false PrintBackground:false Scale:1 PaperWidth:64.86 PaperHeight:11 MarginTop:0.39 MarginBottom:0.39 MarginLeft:0.39 MarginRight:0.39 PageRanges: HeaderTemplate: FooterTemplate: PreferCSSPageSize:false TransferMode:ReturnAsStream}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761801.590705,"logger":"api.formschromiumconverthtml.browser","msg":"-> {\"id\":23,\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\",\"method\":\"Page.printToPDF\",\"params\":{\"scale\":1,\"paperWidth\":64.86,\"paperHeight\":11,\"marginTop\":0.39,\"marginBottom\":0.39,\"marginLeft\":0.39,\"marginRight\":0.39,\"transferMode\":\"ReturnAsStream\"}}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761803.2123337,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"method\":\"CSS.styleSheetAdded\",\"params\":{\"header\":{\"styleSheetId\":\"40781.1\",\"frameId\":\"4CF50E3412634D2BE175F3F92FD721CD\",\"sourceURL\":\"\",\"origin\":\"regular\",\"title\":\"\",\"ownerNode\":10,\"disabled\":false,\"isInline\":false,\"isMutable\":false,\"isConstructed\":false,\"startLine\":0,\"startColumn\":0,\"length\":54,\"endLine\":0,\"endColumn\":54}},\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\"}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761813.717353,"logger":"api.formschromiumconverthtml.browser","msg":"<- {\"id\":23,\"error\":{\"code\":-32000,\"message\":\"Printing failed\"},\"sessionId\":\"C866F07A8BC7E44345D364E4E656FD49\"}","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761813.7174938,"logger":"api.formschromiumconverthtml","msg":"remove user profile directory '/tmp/6fd90129-dc7d-4a1c-862a-e00c79a892e4'","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"debug","ts":1675761813.7961354,"logger":"api.formschromiumconverthtml","msg":"'/tmp/2c16e6f3-1167-4644-9fe0-09c3f139ff0a' removed","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49"}
{"level":"error","ts":1675761813.7961807,"logger":"api","msg":"convert HTML to PDF: convert to PDF: chromium PDF: print to PDF: Printing failed (-32000)","trace":"66cc3c26-4f92-4394-aa5d-d57e13054b49","remote_ip":"192.168.240.138","host":"gotenberg.ilens-core.svc.cluster.local:3000","uri":"/forms/chromium/convert/html","method":"POST","path":"/forms/chromium/convert/html","referer":"","user_agent":"python-httpx/0.23.1","status":500,"latency":21265169486,"latency_human":"21.265169486s","bytes_in":2363489,"bytes_out":21}
{"level":"debug","ts":1675761813.901054,"logger":"api.formschromiumconverthtml","msg":"form data values: map[paperWidth:[64.86]]","trace":"96f4e168-5480-495f-a6ff-26e365b595fe"}`

**Comment by noureentaj** [comment]
Hey! Thanks for the quick reply

My width is bound to scale as table columns increase, so my page transforms
into landscape-like view.
I have an html that has to be converted into pdf. At this point I am not
familiar how I can split it into multiple pages and then merge it into a
PDF. It's just a conversion that's happening at once. I have tried
waitdelay, increasing memory, timeout, everything. What am I missing here?

**Comment by gulien** [comment]
Not much I can do there. I'm updating Chromium in most if not all Gotenberg new release.

**Comment by unstarved** [comment]
Hey team is there any news about this bug?

Just like @QuentinChaumont suggested, I was able to generate big PDF using [thecodingmachine/gotenberg:6.2](https://hub.docker.com/layers/thecodingmachine/gotenberg/6.2/images/sha256-e01f1124ffd4db0cd2387c82d84150a9d14dd8e4538a873398f0faa375eaa14c) which normally fails on `gotenberg:8`.

**Comment by gulien** [comment]
@unstarved I'll need a reproductible scenario to understand what's going on.

**Comment by unstarved** [comment]
Hey @gulien thanks for your patience. I've prepared an input HTML so you can test this locally. The ZIP file contains an `index.html` file that is 10MB uncompressed.

## Test in Gotenberg 8.7

Spin up container
```
docker run --rm -p 3000:3000 gotenberg/gotenberg:8.7  \
  gotenberg \
  --api-port=3000 \
  --api-timeout=300s \
  --api-trace-header=X-Qonto-Correlation-Id \
  --gotenberg-graceful-shutdown-duration=10s \
  --log-format=json \
  --log-level=info \
  --api-disable-health-check-logging=true \
  --prometheus-disable-collect=true \
  --prometheus-disable-route-logging=true \
  --chromium-auto-start=true \
  --libreoffice-disable-routes=true \
  --pdfengines-disable-routes=true \
  --webhook-disable=true
```

Request ends in `500 Internal Server Error`
```
curl -v \
--request POST http://localhost:3000/forms/chromium/convert/html \
--form files=@/tmp/index.html \
-o /tmp/result.pdf
```

Log
```json
{
  "level": "error",
  "ts": 1729154829.3863435,
  "logger": "api",
  "msg": "convert HTML to PDF: convert to PDF: handle tasks: print to PDF: Printing failed (-32000)",
  "trace": "b11deb3c-b601-4af6-8368-cdf8bb0c0f2a",
  "remote_ip": "192.168.215.1",
  "host": "localhost:3000",
  "uri": "/forms/chromium/convert/html",
  "method": "POST",
  "path": "/forms/chromium/convert/html",
  "referer": "",
  "user_agent": "curl/8.7.1",
  "status": 500,
  "latency": 18691845043,
  "latency_human": "18.691845043s",
  "bytes_in": 10410191,
  "bytes_out": 21
}
```

## Test in Gotenberg 6

Spin up container
```
docker run --rm -p 3000:3000 -e MAXIMUM_WAIT_TIMEOUT=120 thecodingmachine/gotenberg:6
```

Request
```
curl -v \
  --request POST http://localhost:3000/convert/html \
  --header 'Content-Type: multipart/form-data' \
  --form files=@/tmp/index.html \
  --form waitTimeout=99.0 \
  -o /tmp/result.pdf
```

It will suceeed to create the PDF (a ~2MB file).

[index.html.zip](https://github.com/user-attachments/files/17409263/index.html.zip)

**Comment by gulien** [comment]
Thanks @unstarved!

I guess you are running Gotenberg on your own machine? Because a common suggestion is to increase the memory available to Gotenberg 🤔 

Could you show me the complete debug logs?

On a side note, are you running Gotenberg on macOS?

**Comment by unstarved** [comment]
- Here I attach the debug logs.
- The container has +20GB assigned but it just breaks at 12GB.
- I'm not sure if it matters whether I run this in macOS or not because I'm user Docker. (but FYI yes, I'm using Orbstack and Macbook).

[debug-logs.txt](https://github.com/user-attachments/files/17409791/debug-logs.txt)

**Comment by gulien** [comment]
I don't see anything in the log that could explain the issue. Chrome seems to crash for an unknown reason 🤔 

Also, there are many requests in these logs?

> I'm not sure if it matters whether I run this in macOS or not because I'm user Docker. (but FYI yes, I'm using Orbstack and Macbook).

Yep, sorry, I was more asking about the architecture (`amd64` vs `arm64`) 😄 

If you're on `arm64`, the `thecodingmachine/gotenberg:6` container is running on an emulated `amd64` platform (thanks to Rosetta), while the `gotenberg/gotenberg:8.7` container is running natively on `arm64`. Dunno if it has an impact though.

Also, why not `gotenberg:gotenberg:8.12`?

**Comment by gulien** [comment]
If you want to stay on `8.7`, could you try with the form field `skipNetworkIdleEvent=true`?

**Comment by unstarved** [comment]
Hey 👋 , updates:
- I've tried with `skipNetworkIdleEvent=true` and I have the same error.
- I've tried with 8.12 (both with and without `skipNetworkIdleEvent=true`) and I have the same error.
- I forgot to mention this but I think this is due to [an old Chromium bug](https://issues.chromium.org/issues/40842550) that is still open (I saw that reference in [this other gotenberg issue](https://github.com/gotenberg/gotenberg/issues/788#issuecomment-1933576620)).

Did you try to run this locally yourself like I explained above? I bet you'll have the same issue.

**Comment by gulien** [comment]
No I did not, but I'm pretty sure I'll have the same issue 👍

**Comment by unstarved** [comment]
OK so what would be the next steps? If it's the Chromium bug there's not much to do here? Is waiting (and praying) the only alternative 🫠 ?

**Comment by gulien** [comment]
Honestly I dunno, there isn't much I can do here, as Chromium seems to be the culprit. You may stick to version 6 in the meantime, but I know it's not ideal...

**Comment by gulien** [comment]
I've added a link in the documentation to this Chromium issue 👍 

Thanks for taking the time to make this subject progress (I kinda have a limited amount of free time currently 😅).

**Comment by boris-szl** [comment]
Any news on this one? Did Chromium fix the issue?
